hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFormatReader.java

```diff
@@ -104,7 +104,7 @@ public boolean hasNext() {
         throw new HoodieIOException("unable to initialize read with log file ", io);
       }
       LOG.info("Moving to the next reader for logfile " + currentReader.getLogFile());
-      return this.currentReader.hasNext();
+      return this.currentReader.hasNext() || hasNext();
```

Comment:
can we just call hasNext() here to simplify?

Reply:
1. When 'spark.speculation' is enabled, two tasks trying to append one log file, the second one will create a new logFile because it can't get lease of file. The second task will leave one zero-size log file when the first task succeed.